client/src/components/Tea/TeaList/TeaListContainer.jsx

```diff
@@ -4,23 +4,82 @@ import { deleteTea, getTeas } from "../../../actions/teaActions";
 import TeaList from "./TeaList";
 
 export class TeaListContainer extends React.Component {
+  state = {
+    sortedIDs: [],
+    memoizedIDs: {},
+    sortColumn: "",
+    sortOrder: ""
+  };
+
   handleDeleteClick = tea => {
     this.props.handleDelete(tea);
   };
 
+  handleSortClick = (columnName, sortOrder) => {
```

Comment:
I like it.  Nice job with the memoization.  I think you can reduce the amount of work here by returning the `el` (which is the tea id) instead of the index during your first `map`.  Then when you sort the mapped array of objects, you already have access to the teaID itself and don't have to cross-reference the index with the original list.